### PR TITLE
TF-2533 [PART-2] Remove email detail iframe dependencies

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2533 

### Dependency PR
- https://github.com/linagora/html-editor-enhanced/pull/34

### View email & Reply in composer
**Before**

https://github.com/linagora/tmail-flutter/assets/160106668/d10eed44-594d-4de6-a1fa-2e9c30a64929

**After**

https://github.com/linagora/tmail-flutter/assets/160106668/a5bdfcb4-6ddc-4c25-941f-febc43101457

### CODE REVIEW

File: core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart [hunk 1]
```diff
@@ -184,53 +228,7 @@ class _HtmlContentViewerOnWebState extends State<HtmlContentViewerOnWeb> {
       ..style.border = 'none'
       ..style.overflow = 'hidden'
       ..style.width = '100%'
-      ..style.height = '100%'
-      ..onLoad.listen((event) async {
```
Comment (hoangdat) [inline core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart:188]: final subscription = iframe.onLoad.listen();
// ....


subscription.cancel();

Comment (tddang-linagora) [reply core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart:188]: Leak still happens this way

**Comment by hoangdat** [comment]
- [ ] test with some html content, which have image lazy load
- [ ] test with some big html content (promotion)

File: core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart [hunk 2]
```diff
@@ -52,7 +52,9 @@ class _HtmlContentViewerOnWebState extends State<HtmlContentViewerOnWeb> {
   String? _htmlData;
   bool _isLoading = true;
   double minHeight = 100;
-  final _jsonEncoder = const JsonEncoder();
+  late final StreamSubscription<html.MessageEvent> sizeListener;
+  bool iframeLoaded = false;
+  static const iframeOnLoadMessage = 'iframeHasBeenLoaded';
```
Comment (dab246) [inline core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart:57]:
```suggestion
  static const String iframeOnLoadMessage = 'iframeHasBeenLoaded';
```

File: core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart [hunk 3]
```diff
@@ -52,7 +52,9 @@ class _HtmlContentViewerOnWebState extends State<HtmlContentViewerOnWeb> {
   String? _htmlData;
   bool _isLoading = true;
   double minHeight = 100;
-  final _jsonEncoder = const JsonEncoder();
+  late final StreamSubscription<html.MessageEvent> sizeListener;
+  bool iframeLoaded = false;
```
Comment (dab246) [inline core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart:56]:
```suggestion
  bool _iframeLoaded = false;
```

File: core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart (hunk 2 above)
Comment (tddang-linagora) [reply core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart]: Done

File: core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart (hunk 3 above)
Comment (tddang-linagora) [reply core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart]: Done

File: lib/features/composer/presentation/composer_controller.dart [hunk 4]
```diff
@@ -219,6 +219,8 @@ class ComposerController extends BaseController {
     _initTextEditor = null;
     _textEditorWeb = null;
     viewState.value = Right(UIClosedState());
+    composerArguments.value = null;
+    emailContentsViewState.value = Right(UIClosedState());
```
Comment (dab246) [inline lib/features/composer/presentation/composer_controller.dart:223]: Also reset `identitySelected`

**Review by dab246 — APPROVED** [review]
- [x] Squash & rebase

**Review by hoangdat — APPROVED** [review]

**Pull request merged**